test/pool/test_max_pool2d_fwd.cpp

```diff
@@ -49,15 +49,20 @@ class TestMaxPool2dFwd : public ::testing::Test
 using true_t  = std::integral_constant<bool, true>;
 using false_t = std::integral_constant<bool, false>;
 
-using KernelTypes = std::conditional_t<CK_ENABLE_FP16 && CK_ENABLE_BF16,
-                                       ::testing::Types<std::tuple<F16, F16, F32, I32, true_t>,
-                                                        std::tuple<F16, F16, F32, I32, false_t>,
-                                                        std::tuple<BF16, BF16, F32, I32, true_t>,
-                                                        std::tuple<BF16, BF16, F32, I32, false_t>,
-                                                        std::tuple<F32, F32, F32, I32, true_t>,
-                                                        std::tuple<F32, F32, F32, I32, false_t>>,
-                                       ::testing::Types<std::tuple<F32, F32, F32, I32, true_t>,
-                                                        std::tuple<F32, F32, F32, I32, false_t>>>;
+using KernelTypes =
+    std::conditional_t<CK_ENABLE_FP16 && CK_ENABLE_BF16 && CK_ENABLE_INT8 && CK_ENABLE_FP8,
```

Comment:
Why do we assume that all types need to be enabled to test int8, for instance? Shouldn't all types to be treated as a separate type?

Reply:
Yes, they generally should be treated separately, but we wanted to omit using `#ifdefs` by using `std::conditional_t` and I can't see any other solutions which would be type specifics than a ladder of ifs so I just left it as two options

Reply:
I have split the tests into separate classes based on type as you suggested